### ISSUE TITLE
Video player black screen and flashing with some twitch streams

### ISSUE THREAD

**Issue by dolwup** [RenderHeads/UnityPlugin-AVProVideo#940]
**Describe the issue**
In VRChat playing certain Twitch streams with AVPro the player screen stays black for the most part and flashes a single frame every one to two seconds. I found another issue (#697)  that was closed because it was stale that refers to exactly the same problem and contains some more detailed information.

**Your Setup (please complete the following information):**
 - Unity version: 2019.4.30
 - AVPro Video version: 2.1.5
 - Operating system vers.ion: Windows 10 64bit (21H1)
 - Device model: 
   - CPU: Intel Core i7 6700K
   - GPU: AMD Radeon RX 6900 XT
 - Video specs (resolution, frame-rate, codec, file size): 
   - Twitch streams of various resolutions and framerates

**To Reproduce**
1. Use an AMD GPU
2. Try to play different Twitch streams on a video player in VRChat (for example this world: https://vrchat.com/home/world/wrld_f8f1cedf-8789-49e8-b846-1a58d30c42ea)

**Logs**
- VRChat log of trying to play a stream on a video player:  [output_log_00-59-24.txt](https://github.com/RenderHeads/UnityPlugin-AVProVideo/files/7229911/output_log_00-59-24.txt)

**Videos**
https://www.youtube.com/watch?v=YGAs_ljloTw

**Comment by kahnivore** [comment]
The first thing to try would be to update to the latest version, 2.2.1, and disable 'stereo support' and 'hap/notchlc support' - this could make a difference. 

If this doesn't help, could you let us know what your GPU usage is looking like in the Task Manager? 

Is the issue only present with Twitch streams?

**Comment by techanon** [comment]
This issue only happens when MediaFoundataion + LowLatency + Hardware decoding is enabled for the video player and only for AMD (afaict)

Here's another testing video with just the trial package of the latest release (2.2.1) https://streamable.com/ki0erj
The URL for the twitch video was retrieved via youtubeDL separately. 

All other statements I have made in #697 are still accurate.

Changing Stereo Support and HAP/Notchic support did not make any difference.

You can see in the video that when I swapped to WinRT video API, the hardware decoding + low latency appeared to work as expected, but VRChat does not expose this setting to world developers currently. I will be bringing this up with them for further discussion and options as they currently seem to default internally to Media Foundation video api and hardware decoding options.

**Comment by techanon** [comment]
Additionally here are two other testing videos:

dlive.tv : https://streamable.com/1nvlrw
youtube.com : https://streamable.com/98exkm

TLDR; as said in #697, twitch and dlive both have the same issue, but youtube seems to not on the vast majority of cases.

**Comment by techanon** [comment]
Also, could I get some clarification on the origins of this comment? In what way does it not like unity audio?

https://www.renderheads.com/content/docs/AVProVideo/articles/component-media-player.html

> WinRT is the new Windows API and has better support for adaptive streaming media, but doesn't support many other features (eg Unity audio)

**Comment by kahnivore** [comment]
@techanon with regards to WinRT and audio - there is no way to grab the audio, it can only be sent straight to the speakers, so we can't implement sending audio to Unity, only system direct. 

Could you try updating to the latest 2.2.2? There have been some bug fixes that might help. 

Wondering also if twitch has some sort of content protection on some of its streams...

**Comment by techanon** [comment]
Here is a video demonstrating the results of different streams with different format ids used (ie: resolutions)
https://streamable.com/38hfgy
All tests used MediaFoundation + Hardware Encoding + Low Latency
Tried 3 different channels (for source media variety) with different formats.
Interestingly the rocket league channel had a decently acceptable time with the decoding process, whereas the other two have the single frame flash issue.

The order is as follows:
Channel: RocketLeague
- 360p
- 720p60

Stream information extracted via VLC for 720p60
![image](https://user-images.githubusercontent.com/1837484/138206745-64f7ac99-1a26-4810-925c-a2540edc967f.png)


Channel: Ironmouse
- 360p
- 160p
- 720p

Stream information extracted via VLC for 720p
![image](https://user-images.githubusercontent.com/1837484/138206776-aba24667-8e14-46b3-802b-ad11c40f6f9b.png)


Channel: ESL_CSGO
- 360p
- 1080p60__source_

Stream information extracted via VLC for 1080p60__source_
![image](https://user-images.githubusercontent.com/1837484/138206805-9ce4f04d-b72c-40f8-81a7-a3d4a9ed8aa5.png)

The fact that the rocket league one works so well (only occasional green smears) even on high res puzzles me.

Is there any other kind of information that would be useful in understanding the difference of results between streams?
